src/Features/CSharp/Portable/CodeRefactorings/MoveDeclarationNearReference/MoveDeclarationNearReferenceCodeRefactoringProvider.cs

```diff
@@ -174,7 +174,7 @@ private StatementSyntax CreateMergedDeclarationStatement(State state, StatementS
                 // Type inference.  Only merge if types match.
                 var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
                 var rightType = semanticModel.GetTypeInfo(expression.Right, cancellationToken);
-                return (localSymbol.Type == null && rightType.Type == null) || localSymbol.Type.Equals(rightType.Type);
+                return Equals(localSymbol.Type, rightType.Type);
```

Comment:
Wow, I kind of hate this.  Would seem much clearer to me to have `object.Equals` in this case.